google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -919,6 +1245,24 @@ Session getReadWriteSession() {
     return sess;
   }
 
+  private PooledSession replaceReadSession(SessionNotFoundException e, PooledSession session) {
+    if (!options.isFailIfSessionNotFound() && session.allowReplacing) {
+      closeSessionAsync(session);
+      return getReadSession();
+    } else {
+      throw e;
+    }
+  }
+
+  PooledSession replaceReadWriteSession(SessionNotFoundException e, PooledSession session) {
```

Comment:
should these have the same visibility?

Reply:
`replaceReadWriteSession` needs at least package visibility, as it is called from `DatabaseClientImpl`. `replaceReadSession` does not need more than private visibility, as it is only called from withing `SessionPool.java`.